### ISSUE TITLE
Tomcat default args.vm.override.string results in fatal errors on startup

### ISSUE THREAD

**Issue by jfrantzius** [redhat-developer/rsp-server-community#167]
Hi,
when creating a Tomcat server, the default `args.vm.override.string` found in the server config doesn't work. There are two errors on server startup, preventing it from coming up.

First error:
```
Configuration error
java.io.FileNotFoundException: /conf/logging.properties (No such file or directory)
```
This is because the `args.vm.override.string` contains `-Djava.util.logging.config.file=/conf/logging.properties` 

Second error:
```
Feb. 14, 2023 9:31:22 PM org.apache.catalina.startup.Catalina parseServerXml
WARNUNG: Unable to load server configuration from [/Users/joerg.frantzius/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.0.8/apache-tomcat-10.0.8/bin/conf/server.xml]
java.io.FileNotFoundException: /Users/joerg.frantzius/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.0.8/apache-tomcat-10.0.8/bin/conf/server.xml (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
```
This looks like `/bin` is appended programmatically either to `server.base.dir` or `server.home.dir` somewhere.

Steps to reproduce:
1. Using "Community Server Connectors" v0.25.7
2. Create a Tomcat server (both 9.x and 10.x showed the problem)
3. Click "Edit server"
4. Set `"args.override.boolean": "true"`
5. Start the server

**Comment by hayden-mumm** [comment]
I am having this exact same issue

**Comment by hayden-mumm** [comment]
Seems likely related to [server.json](https://github.com/redhat-developer/rsp-server-community/blob/master/rsp/runtimes/bundles/org.jboss.tools.rsp.server.tomcat/src/main/resources/servers.json#L302) workingDirectory being set to bin. Only reference to bin I could find.

Also it seems to be using ${server.base.dir} here inside of vmArgs, that is what it should be doing for the default server.config as well, however that doesn't work currently. (says java.io.FileNotFoundException: '${server.base.dir}\conf\logging.properties' (The system cannot find the path specified))

**Comment by robstryker** [comment]
When i create a new server, and use the download feature instead of the on-disk option, I get the following server json:
```
{
  "args.override.boolean": "false",
  "args.program.override.string": "start",
  "args.vm.override.string": "-Djava.util.logging.config.file='/conf/logging.properties' -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager -Djdk.tls.ephemeralDHKeySize=2048 -Djava.protocol.handler.pkgs=org.apache.catalina.webresources -Dorg.apache.catalina.security.SecurityListener.UMASK=0027 -Dcatalina.base='' -Dcatalina.home='/home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4' -Djava.io.tmpdir='/temp'",
  "id": "apache-tomcat-10.1.4",
  "id-set": "true",
  "org.jboss.tools.rsp.server.typeId": "org.jboss.ide.eclipse.as.server.tomcat.100",
  "server.base.dir": "/home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4",
  "server.deploy.dir": "${server.base.dir}/webapps/",
  "server.home.dir": "/home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4",
  "server.http.host": "localhost",
  "server.http.port": "8080",
  "vm.install.path": ""
}```

Part of the output when I start the server includes:

```
CATALINA_BASE:         /home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4
22-Feb-2023 12:33:24.090 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_HOME:         /home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4
22-Feb-2023 12:33:24.096 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.config.file=/home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4/conf/logging.properties
```


The launched command, as fetched by `ps -ax | grep tomcat` is as follows:

```
[rob@fedora apache-tomcat-10.1.4]$ ps ax | grep tomcat
 905293 ?        Sl     0:05 /home/rob/apps/java/java17/oracle/jdk-17.0.1/bin/java -Djava.util.logging.config.file=/home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4/conf/logging.properties -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager -Djdk.tls.ephemeralDHKeySize=2048 -Djava.protocol.handler.pkgs=org.apache.catalina.webresources -Dorg.apache.catalina.security.SecurityListener.UMASK=0027 -Dcatalina.base=/home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4 -Dcatalina.home=/home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4 -Djava.io.tmpdir=/home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4/temp -classpath /home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4/bin/bootstrap.jar:/home/rob/.rsp/redhat-community-server-connector/runtimes/installations/tomcat-10.1.4_3/apache-tomcat-10.1.4/bin/tomcat-juli.jar org.apache.catalina.startup.Bootstrap start
```


All of these paths appear to exist for me. The server starts up without error. 

One thing I do note is that the generated args.vm.override.string appears to be missing the basedir / full path. 

tl;dr is that I can't replicate, but I'm looking into it? Any more info you all can provide would be very helpful, like what the launched process command actually looks like

**Comment by hayden-mumm** [comment]
I'm running this on windows so I'm not sure where to see what command started it. I am also using Tomcat 9.0.41 instead of tomcat 10 if that makes a difference.

I did download this with the plugin, it's not using an existing tomcat install.

If you walk me through how to get the information you need I'd be happy to provide it.

**Comment by robstryker** [comment]
I made a release today and I'm not sure if it will fix your issue. I THINK it should but if it doesn't, please re-open this.

**Comment by jfrantzius** [comment]
I saw a new release for the Community Server Extension in VS Code, which told me to reload. After reload, the extension still has v0.25.7, though? Server Connector is at v0.26.1.

I still see the errors, with these steps to reproduce:

1. Create new server, with Tomcat 10.0.8 via download
2. Deployed a webapp, see that it succeeds starting up, stop the server
3. Clicked "Edit Server", change `args.override.boolean` to true
4. Start server again, still getting the same error messages as above

Is there perhaps a way to use a newer version of the Community Server Extension in VS Code?

**Comment by jfrantzius** [comment]
Not being sure how the release process works and how the extension is made available in the extensions marketplace, but there seem to be failures for Mac and Windows in https://github.com/redhat-developer/rsp-server-community/actions/runs/4246235706

**Comment by robstryker** [comment]
It appears it was an intermittent failure due to downloading of vscode failing.

**Comment by robstryker** [comment]
There was an unknown error when sending it to marketplace. Seems marketplace thought 0.26.1 was there but never displayed it. Very weird. 

Anyway I've upversioned and sent 0.26.2 and it seems to have worked.

**Comment by robstryker** [comment]
So the deploy seemed to work but its still not showing up on marketplace. Very strange.
